modules/cli/src/main/java/org/apache/ignite/cli/IgnitePaths.java

```diff
@@ -20,43 +20,96 @@
 import java.io.File;
 import java.nio.file.Path;
 
+/**
+ * The main resolver of Ignite paths for the current installation (like bin, work and etc. dirs).
+ * Current Ignite distributive has the following dirs structure:
+ * <ul>
+ *     <li>bin</li>
+ *     <ul>
+ *         <li>${version}</li>
+ *         <ul>
+ *             <li>cli</li>
+ *             <li>libs</li>
+ *         </ul>
+ *     </ul>
+ *     <li>work</li>
+ *     <ul>
+ *         <li>config</li>
+ *         <ul>
+ *             <li>default-config.xml</li>
+ *         </ul>
+ *         <li>cli</li>
+ *         <ul>
+ *             <li>pids</li>
+ *         </ul>
+ *         <li>modules.json</li>
+ *     </ul>
+ * </ul>
+ */
 public class IgnitePaths {
-
+    /**
+     * Path to Ignite bin directory.
+     * Bin directory contains jar artifacts for Ignite server and CLI modules.
+     */
     public final Path binDir;
+
+    /**
+     * Work directory for Ignite server and CLI operation.
+     */
     public final Path workDir;
-    private final String version;
 
-    public IgnitePaths(Path binDir, Path workDir, String version) {
+    /**
+     * Ignite CLI version.
+     * Also, the same version will be used for addressing any binaries inside bin dir
+     */
+    private final String ver;
+
+    /**
+     * Creates resolved ignite paths instance from Ignite CLI version and base dirs paths.
+     *
+     * @param binDir Bin directory.
+     * @param workDir Work directory.
+     * @param ver Ignite CLI version.
+     */
+    public IgnitePaths(Path binDir, Path workDir, String ver) {
         this.binDir = binDir;
         this.workDir = workDir;
-        this.version = version;
+        this.ver = ver;
     }
 
-
+    /** Path where CLI module artifacts will be placed. */
```

Comment:
One-line Javadoc comments allowed and preferred for fields. I'm not sure that this can be applied to methods.